pkg/client/common_test.go

```diff
@@ -56,46 +64,83 @@ func testCommon(t *testing.T, when spec.G, it spec.S) {
 					},
 				},
 			}
-			accessChecker = ifakes.NewFakeAccessChecker()
 		})
 
 		when("passed specific run image", func() {
+			it.Before(func() {
+				publish = false
+			})
+
 			it("selects that run image", func() {
 				runImgFlag := "flag/passed-run-image"
-				runImageName := subject.resolveRunImage(runImgFlag, defaultRegistry, "", stackInfo.RunImage, nil, false, accessChecker)
+				runImageName = subject.resolveRunImage(runImgFlag, defaultRegistry, "", stackInfo.RunImage, nil, publish)
 				assert.Equal(runImageName, runImgFlag)
 			})
 		})
 
 		when("publish is true", func() {
-			it("defaults to run-image in registry publishing to", func() {
-				runImageName := subject.resolveRunImage("", gcrRegistry, defaultRegistry, stackInfo.RunImage, nil, true, accessChecker)
-				assert.Equal(runImageName, gcrRunMirror)
-			})
-
-			it("prefers config defined run image mirror to stack defined run image mirror", func() {
-				configMirrors := map[string][]string{
-					runImageName: {defaultRegistry + "/unique-run-img"},
-				}
-				runImageName := subject.resolveRunImage("", defaultRegistry, "", stackInfo.RunImage, configMirrors, true, accessChecker)
-				assert.NotEqual(runImageName, defaultMirror)
-				assert.Equal(runImageName, defaultRegistry+"/unique-run-img")
+			when("desirable run-image are accessible", func() {
```

Comment:
I could be wrong, but I'm not sure if these tests need to care if publish is true or not. The access checker is abstracting away that part (via the pull policy).

Reply:
Yeah, we are using mocks and the parameters to the method are not relevant in these tests, I removed some of them